examples/proxy/envelope-encryption/PREPARE_KMS.md

```diff
@@ -0,0 +1,123 @@
+# Preparing a Key Management Service (KMS)
+
+As discussed [here](./README.md) you must provide a KMS solution in order to use the Envelope Encryption Filter.
+For Tech Preview, only HashiCorp Vault is supported.
+
+## Preparing HashiCorp Vault
+
+You may either use an existing instance of HashiCorp Vault you have available in your organisation, or deploy a development instance
+HashoCorp Vault to your OpenShift Cluster which is good enough to try out the Filter.
+
+## Using an Existing Vault Instance
+
+### Prerequistes
+
+You must have an HashCorp Vault instance running on premise or you have a HashiCorp HCP Vault or Enterprise.  Free tier HCP Vault Secrets is
+*not* adequate.
+
+* You must have adminstrative access to the Vault instance or Vault Namespace.
+* You must know the Address of the Vault Server, and if Vault Enterprise, the name of the Vault Namespace(s).
+* Vault CLI installed.
+
+Note: these examples are prepared against HashiCorp Vault 1.15.
+
+### Configure the Existing Instance
+
+1. Set the `VAULT_ADDR` (and `VAULT_NAMESPACE` if Enterprise) to refer to the Vault Environment.
+   ```sh
+   export VAULT_ADDR=https://<vault server>:8200
+   export VAULT_NAMESPACE=<namespace(s)>
+2. Login to Vault as the Adminstrator.
+   ```sh
+   vault login
+   ```
+3. Enable the Transit Secrets Engine.
+   ```sh
+   vault secrets enable transit
+   ```
+   If the Transit engine was already enabled, an error will be printed but this can be ignored.
+4. Create a variable containing the Vault Transit Url.  This will be used later on.
+   ```sh
+      export VAULT_TRANSIT_URL=${VAULT_ADDR}/v1/${VAULT_NAMESPACE}/transit
+   ```
+5. Update the proxy config to refer to your Vault instance:
```

Comment:
FYI If we get https://github.com/kroxylicious/kroxylicious/pull/964 done, these `sed` steps will go away.  Instead, we'll create a config map here.  That CM will be mounted into the pod, and an `envFrom` to pull the CM value into the pod's environment.  Then the interpolation will put the value into the file.